run.go

```diff
@@ -326,38 +329,96 @@ func Log(c chan *SimulationStatus) DomainManipulator {
 	}
 }
 
-func (d *InMAP) checkOutputNames(n ...string) error {
+func removeDuplicates(s []string) []string {
+	result := []string{}
+	seen := map[string]string{}
+	for _, val := range s {
+		if _, ok := seen[val]; !ok {
+			result = append(result, val)
+			seen[val] = val
+		}
+	}
+	return result
+}
+
+// Checks if outputVariables are defined in terms of other outputVariables.
+// Creates a slice of unique model variables that are required to calculate
+// the requested outputVariables.
+func checkForDerivitives(m map[string]string) (map[string]string, []string) {
+	getVariables := []string{}
```

Comment:
I think there are cases in which the number of variables that we need to retrieve from the model could be less than the number of output variables defined by the user. For example, there could be three output variables that all are the product of different calculations involving Windspeed (Windspeed, Windspeed^2, Windspeed^3). Does this make sense?